WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt

```diff
@@ -781,6 +808,58 @@ class PagesViewModel
         }
     }
 
+    fun updateAuthorFilterSelection(selectionId: Long) {
+        val selection = AuthorFilterSelection.fromId(selectionId)
+
+        updateViewStateTriggerPagerChange(
+                authorFilterSelection = selection,
+                authorFilterItems = getAuthorFilterItems(selection, accountStore.account?.avatarUrl)
+        )
+        if (isFilteringByAuthorSupported) {
+            prefs.postListAuthorSelection = selection
+        }
+    }
+
+    /**
+     * Filtering by author is disable on:
+     * 1) Self-hosted sites - The XMLRPC api doesn't support filtering by author.
+     * 2) Jetpack sites - we need to pass in the self-hosted user id to be able to filter for authors
+     * which we currently can't
+     * 3) Sites on which the user doesn't have permissions to edit posts of other users.
+     *
+     * This behavior is consistent with Calypso and Posts as of 11/4/2019.
+     */
+    private val isFilteringByAuthorSupported: Boolean by lazy {
+        site.isWPCom && site.hasCapabilityEditOthersPages
+    }
+
+    private fun updateViewStateTriggerPagerChange(
+        isAuthorFilterVisible: Boolean? = null,
+        authorFilterSelection: AuthorFilterSelection? = null,
+        authorFilterItems: List<AuthorFilterListItemUIState>? = null
+    ) {
+        val currentState = requireNotNull(authorUIState.value) {
+            "updateViewStateTriggerPagerChange should not be called before the initial state is set"
+        }
+
+        _authorUiState.value = PagesAuthorFilterUIState(
+                isAuthorFilterVisible ?: currentState.isAuthorFilterVisible,
+                authorFilterSelection ?: currentState.authorFilterSelection,
+                authorFilterItems ?: currentState.authorFilterItems
+        )
```

Comment:
Np: Can we use `currentState.copy` instead of creating a new `PagesAuthorFilterUIState` object? Feel free to ignore this comment as I understand you just reused the logic from `PostListMainViewModel`.

Reply:
Great suggestion. Dropped it in the next subtask PR. Thanks!